qa/task-cancellation/src/main/java/org/elasticsearch/plugin/tasks/TransportTestAction.java

```diff
@@ -0,0 +1,151 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.tasks;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionListenerResponseHandler;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.GroupedActionListener;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.util.concurrent.AbstractRunnable;
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskCancelledException;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.Transport;
+import org.elasticsearch.transport.TransportActionProxy;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+public class TransportTestAction extends HandledTransportAction<TestRequest, TestResponse> {
+    static final Logger logger = LogManager.getLogger(TransportTestAction.class);
+    static final ActionType<TestResponse> ACTION = new ActionType<>("internal::test_action", TestResponse::new);
+    final TransportService transportService;
+    final Map<String, CountDownLatch> latches = ConcurrentCollections.newConcurrentMap();
+    final ClusterService clusterService;
+
+    @Inject
+    public TransportTestAction(TransportService transportService, ActionFilters actionFilters, ClusterService clusterService) {
+        super(ACTION.name(), transportService, actionFilters, TestRequest::new, ThreadPool.Names.GENERIC);
+        this.transportService = transportService;
+        this.clusterService = clusterService;
+        TransportActionProxy.registerProxyAction(transportService, ACTION.name(), TestResponse::new);
+    }
+
+    @Override
+    protected void doExecute(Task task, TestRequest request, ActionListener<TestResponse> origListener) {
+        if (request.targets.isEmpty()) {
+            origListener.onFailure(new IllegalArgumentException("rem"));
```

Comment:
why "rem"?

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/59828/commits/c065a980791089b91b577d0bb9df17a2623d54ca